### ISSUE TITLE
no-unused-var misses arguments in function expression assignment

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#697]
The `no-unsed-vars` rule is currently missing patterns like this:

``` js
Foo.bar = function(baz) {
    // ...
};
```

`baz` should be flagged as being unused, but it is not.
